sequence_task_1/student/include/impl.h

```diff
@@ -41,11 +55,71 @@ class StaticPlaylist
     void switchNext();
 
     /** @todo Amount of tracks in playlist */
-    size_type count() const;
+    size_type count() const { return (size_type)std::distance(icurrent, end()); }
```

Comment:
Please, reimplement this task without that additional iterator, try to simplify things. Just use information from day 1 and day2 of the training.